src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -297,14 +299,19 @@ public static void registerFuelHandler(IFuelHandler handler)
     {
         fuelHandlers.add(handler);
     }
+    
     public static int getFuelValue(ItemStack itemStack)
     {
-        int fuelValue = 0;
-        for (IFuelHandler handler : fuelHandlers)
+        ListIterator<IFuelHandler> li = fuelHandlers.listIterator(fuelHandlers.size());
+        while (li.hasPrevious())
         {
-            fuelValue = Math.max(fuelValue, handler.getBurnTime(itemStack));
+            int fuelValue = li.previous().getBurnTime(itemStack);
+            if (fuelValue > 0)
```

Comment:
0 means that it is not your item. see the improved javadoc [here](https://github.com/liachmodded/MinecraftForge/blob/fuel-1.8.8/src/main/java/net/minecraftforge/fml/common/IFuelHandler.java#L19-L31).